libs/application/templates/parental-leave/src/fields/Rights/RequestDaysSlider.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { FC, useState } from 'react'
+import { FieldBaseProps, getValueViaPath } from '@island.is/application/core'
+import { Controller, useFormContext } from 'react-hook-form'
+import { Box, Text } from '@island.is/island-ui/core'
+import { theme } from '@island.is/island-ui/theme'
+import { useLocale } from '@island.is/localization'
+import { m } from '../../lib/messages'
+import Slider from '../components/Slider'
+import BoxChart, { BoxChartKey } from '../components/BoxChart'
+import config from '../../config'
+
+const RequestDaysSlider: FC<FieldBaseProps> = ({ field, application }) => {
+  const maxDays = config.maxDaysToGiveOrReceive
+  const { id } = field
+  const { formatMessage } = useLocale()
+  const currentAnswer = getValueViaPath(
+    application.answers,
+    field.id,
+    1,
+  ) as number
+
+  const { clearErrors } = useFormContext()
+
+  const [chosenRequestDays, setChosenRequestDays] = useState<number>(
+    currentAnswer,
+  )
+
+  const daysStringKey =
+    chosenRequestDays > 1 ? m.requestRightsDays : m.requestRightsDay
+
+  const boxChartKeys: BoxChartKey[] = [
+    {
+      label: () => ({
+        ...m.yourRightsInMonths,
+        values: { months: config.defaultMonths },
+      }),
+      bulletStyle: 'blue',
+    },
+    {
+      label: () => ({ ...daysStringKey, values: { day: chosenRequestDays } }),
+      bulletStyle: 'greenWithLines',
+    },
+  ]
+
+  return (
+    <Box marginBottom={6}>
+      <Text marginBottom={4} variant="h3">
+        {formatMessage(m.requestRightsDaysTitle)}
+      </Text>
+      <Box marginBottom={12}>
+        <Controller
+          defaultValue={chosenRequestDays}
+          name={id}
+          render={({ onChange, value }) => (
+            <Slider
+              label={{
+                singular: formatMessage(m.day),
+                plural: formatMessage(m.days),
+              }}
+              min={1}
+              max={maxDays}
+              step={1}
+              currentIndex={value || chosenRequestDays}
+              showMinMaxLabels
+              showToolTip
+              trackStyle={{ gridTemplateRows: 8 }}
+              calculateCellStyle={() => {
+                return {
+                  background: theme.color.dark200,
+                }
+              }}
+              onChange={(newValue: number) => {
+                clearErrors(id)
+                onChange(newValue)
+                setChosenRequestDays(newValue)
+              }}
+            />
+          )}
+        />
+      </Box>
+      <BoxChart
+        application={application}
+        boxes={config.defaultMonths + 1}
```

Comment:
Might want to have a maxMonths in the config?

Reply:
Agree, fixed now.